code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -227,6 +237,40 @@
 				user << "<span class='warning'>Please close the access panel before locking it.</span>"
 			else
 				user << "<span class='warning'>Access denied.</span>"
+	else if(istype(W, /obj/item/device/paicard))
+		if(paicard)
+			user << "<span class='warning'>A [paicard] is already inserted!</span>"
+		else if(allow_pai && !key)
+			if(!locked && !open)
+				var/obj/item/device/paicard/card = W
+				if(card.pai && card.pai.mind)
+					if(!user.drop_item())
+						return
+					W.forceMove(src)
+					paicard = card
+					user.visible_message("[user] inserts [W] into [src]!","<span class='notice'>You insert [W] into [src].</span>")
+					paicard.pai.mind.transfer_to(src)
+					src << "<span class='notice'>You sense your form change as you are uploaded into [src].</span>"
+					bot_name = name
+					name = paicard.pai.name
+					pai_history += text("\[[time_stamp()]\] [paicard.pai][(ismob(paicard.pai) && paicard.pai.ckey) ? "([paicard.pai.ckey])" : ""] added to [src.bot_name] by [user][(ismob(user) && user.ckey) ? "([user.ckey])" : ""]")
+				else
+					user << "<span class='warning'>[W] is inactive.</span>"
+			else
+				user << "<span class='warning'>The personality slot is locked.</span>"
+		else
+			user << "<span class='warning'>[src] is not compatible with [W]</span>"
+	else if (istype(W, /obj/item/weapon/hemostat) && paicard)
+		if (open)
+			user << "<span class='warning'>Close the access panel before manipulating the personality slot!</span>"
+		else
+			var/T = user.loc
+			user << "<span class='notice'>You attempt to pull [paicard] free...</span>"
+			if(do_after(user, 30, target = src))
+				if(istype(src, /mob/living/simple_animal/bot) && user && W && T)
+					if(user.get_active_hand() == W && user.loc == T) //The insanity of sanity checks, folks.

```

Comment:
you don't need the second check here as it would fail do_after
just merge those checks down into this
`if(W && user.get_active_hand() == W)`
and remove the `var/T = user.loc` above as well since it's not needed